main.c

```diff
@@ -116,6 +121,7 @@ main(int argc, char *argv[])
 	struct scryptdec_file_cookie * C = NULL;
 	enum passphrase_entry passphrase_entry = PASSPHRASE_UNSET;
 	const char * passphrase_arg;
+	char * passwd_env;
```

Comment:
Even though `getenv` returns a `char *`, I would make this `const char *` in order to enforce the "can't be modified" constraint.